doc/diagnostics/AddingTracepoints.md

```diff
@@ -0,0 +1,417 @@
+
+This requires all newer .tdf files to be supersets of old .tdf files. Ordering in .tdf files governs numbering, starting with the first tracepoint in the file being 0 and incrementing down the file. This is why tracepoints must be added at the end - otherwise each subsequent tracepoint's number will be incremented, and the formatter will attempt to format the tracepoint using the format string of another tracepoint. 
+
+Setting the `Obsolete` keyword guards against usage of the old tracepoint in the current code stream, but keeps it available for the formatter to use when processing trace from older releases.
+
+
+## Adding tracepoints to a module that already has some
+
+Adding a tracepoint to a module that is already enabled for trace requires four steps:
+
+1. Determine the trace module name
+2. Add your tracepoint to .tdf file
+3. Add your tracepoint to your source code
+4. Test the tracepoint
+
+1) First determine the module you are about to add tracepoints to.
+
+The name that OpenJ9 trace uses to identify a module/component can be found in the module's trace definition (.tdf) file.  You should find a .tdf file in your module's root directory.  For example, the .tdf file for the j9vm component is runtime/vm/j9vm.tdf. The first line in that file: `Executable=j9vm` specifies the name that OpenJ9 trace (and -Xtrace options) use to identify tracepoints in that component. The .tdf file contains the specifications for all tracepoints in the j9vm component, including their format strings. Files `ut_j9vm.h` and `ut_j9vm.c`, are generated at build time from the contents of the .tdf.
+
+2) Step two is to add the tracepoint to your .tdf file. All new tracepoints must be added to the end of the .tdf file. This ensures that tracepoints preserve their tracepoint numbers between releases, see rules above.
+
+For example:
+
+`TraceEvent=Trc_VM_getMethodOrFieldID_dereferencedClass Overhead=1 Level=3 Template="class=%p (%.*s)"`
+
+This breaks down as follows:
+
+`TraceEvent=Trc_VM_getMethodOrFieldID_dereferencedClass`  This specifies the type and the name of the tracepoint. `Trc_VM_getMethodOrFieldID_dereferencedClass` is the name of the generated macro to use in your code. The `TraceEvent=` denotes this is an event trace point. Other tracepoint types are:  `TraceEntry` `TraceExit` `TraceException` `TraceExit-Exception` `TraceAssert` `TraceDebug`.
+
+`Overhead=1`  Each tracepoint has an overhead to determine whether it is included in the generated object code at build time. This is useful for very hot tracepoints, where even the cost of the simple 'if enabled' test at runtime is too great. The threshold in the tracepoint pre-processor is currently set to 1, so if you set Overhead to a value greater than 1, the tracepoint will be no-oped out at build time. It can still be useful to have the tracepoint defined in the source code, but of course requires a custom re-build of the relevant library as and when actually needed for diagnostics.
```

Comment:
Suggest 'no-oped' be replaced: 'the tracepoint will expand to an empty statement at build time'.